apps/prairielearn/src/components/CreateQuestionModalContents.tsx

```diff
@@ -0,0 +1,269 @@
+  cardRef?: (el: HTMLElement | null) => void;
+}
+
+function SelectableCard({
+  id,
+  title,
+  description,
+  selected,
+  onClick,
+  onKeyDown,
+  cardRef,
+}: SelectableCardProps) {
+  const handleKeyDown = (e: KeyboardEvent) => {
+    if (e.key === 'Enter' || e.key === ' ') {
+      e.preventDefault();
+      onClick();
+    }
+    onKeyDown?.(e);
+  };
+
+  return (
+    <div
+      ref={cardRef}
+      class={clsx('card h-100', {
+        'border-primary bg-primary bg-opacity-10': selected,
+        'border-secondary': !selected,
+      })}
+      style={{ cursor: 'pointer' }}
+      role="radio"
+      tabIndex={selected ? 0 : -1}
+      aria-checked={selected}
+      aria-labelledby={`${id}_title`}
+      aria-describedby={`${id}_description`}
+      onClick={onClick}
+      onKeyDown={handleKeyDown}
+    >
+      <div class="card-body text-center d-flex flex-column justify-content-center">
+        <div id={`${id}_title`} class={clsx('card-title', { 'text-primary fw-bold': selected })}>
+          {title}
+        </div>
+        <div id={`${id}_description`} class={clsx('card-text small', { 'text-muted': !selected })}>
+          {description}
+        </div>
+      </div>
+      {selected && (
+        <div class="position-absolute top-0 end-0 p-2">
+          <i class="fa fa-check-circle text-primary" aria-hidden="true"></i>
+        </div>
+      )}
+    </div>
+  );
+}
+
+interface RadioCardGroupProps {
+  label: string;
+  value: string;
+  options: { id: string; title: string; description: string }[];
+  onChange: (value: string) => void;
+}
+
+function RadioCardGroup({ label, value, options, onChange }: RadioCardGroupProps) {
+  const cardRefs = useRef<(HTMLElement | null)[]>([]);
+
+  const handleKeyDown = (e: KeyboardEvent, currentIndex: number) => {
+    let newIndex = currentIndex;
+
+    switch (e.key) {
+      case 'ArrowDown':
+      case 'ArrowRight':
+        e.preventDefault();
+        newIndex = (currentIndex + 1) % options.length;
+        break;
+      case 'ArrowUp':
+      case 'ArrowLeft':
+        e.preventDefault();
+        newIndex = currentIndex === 0 ? options.length - 1 : currentIndex - 1;
+        break;
+      case 'Home':
+        e.preventDefault();
+        newIndex = 0;
+        break;
+      case 'End':
+        e.preventDefault();
+        newIndex = options.length - 1;
+        break;
+      default:
+        return;
+    }
+
+    onChange(options[newIndex].id);
+
+    // Focus the newly selected element, but only after a rerender.
+    setTimeout(() => {
+      cardRefs.current[newIndex]?.focus();
+    }, 0);
+  };
+
+  return (
+    <fieldset class="mb-3">
+      {/* `col-form-label` correctly overrides the default font size for `legend` */}
+      <legend class="col-form-label">{label}</legend>
+      <div
+        role="radiogroup"
+        aria-label={label}
+        style={{
+          display: 'grid',
+          gridTemplateColumns: 'repeat(auto-fit, minmax(240px, 1fr))',
```

Comment:
This fixes the issue:

```suggestion
          gridTemplateColumns: 'repeat(auto-fit, minmax(min(240px, 100%), 1fr))',
```


<img width="495" height="1199" alt="Image" src="https://github.com/user-attachments/assets/a3d962a2-07a3-418b-a231-4f6a1af200fa" />

Reply:
Great suggestion! I'll run with this 🙌